src/metabase/driver.clj

```diff
@@ -23,6 +23,14 @@
    to scan millions of values at any rate."
   10000)
 
+(def ^:const field-values-lazy-seq-chunk-size
+  "How many Field values should be fetched at a time for a chunked implementation of `field-values-lazy-seq`?"
+  ;; Hopefully this is a good balance between
+  ;; 1. Not doing too many DB calls
+  ;; 2. Not running out of mem
+  ;; 3. Not fetching too many results for things like mark-json-field! which will fail after the first result that isn't valid JSON
+  500)
```

Comment:
I'm not sure this is something that really needs to be a constant on the driver interface because from driver to driver this may change.  This feels like something that we can just leave up to each individual driver implementation that needs it and keep it scoped there.